hathor/wallet/resources/thin_wallet/send_tokens.py

```diff
@@ -173,32 +193,34 @@ def _responseFailed(self, err, tx, request):
             self.manager.stratum_factory.update_jobs()
         else:
             # if we're mining on a thread, stop it
-            request.should_stop_mining_thread = True
+            context.should_stop_mining_thread = True
 
-    def _stratum_deferred_resolve(self, tx: Transaction, request: Request) -> None:
+    def _stratum_deferred_resolve(self, context: _Context) -> None:
         """ Method called after stratum resolves tx proof of work
             We remove the mining data of this tx on stratum and start a new thread to verify the tx
         """
-        funds_hash = tx.get_funds_hash()
-        tx = self.manager.stratum_factory.mined_txs[funds_hash]
+        funds_hash = context.tx.get_funds_hash()
+        context.tx = self.manager.stratum_factory.mined_txs[funds_hash]
         # Delete it to avoid memory leak
         del(self.manager.stratum_factory.mined_txs[funds_hash])
 
-        deferred = threads.deferToThreadPool(reactor, self.manager.pow_thread_pool, self._stratum_thread_verify, tx)
-        deferred.addCallback(self._cb_tx_resolve, request)
-        deferred.addErrback(self._err_tx_resolve, request, 'stratum_resolve')
+        deferred = threads.deferToThreadPool(reactor, self.manager.pow_thread_pool,
+                                             self._stratum_thread_verify, context)
+        deferred.addCallback(self._cb_tx_resolve)
```

Comment:
No, `_stratum_thread_verify` returns a `_Context`, so it will become the first argument to `_cb_tx_resolve`. I really wish we could use `async/await` and avoid these convoluted functions and argument passing though.